microservices-connector/internal/controller/gmconnector_controller.go

```diff
@@ -404,27 +587,47 @@ func getTemplateBytes(resourceType string) ([]byte, error) {
 	return yamlBytes, nil
 }
 
-func reconcileRouterService(ctx context.Context, client client.Client, graph *mcv1alpha3.GMConnector) error {
-	routerService := &corev1.Service{}
-	jsonBytes, err := json.Marshal(graph)
+func (r *GMConnectorReconciler) reconcileRouterService(ctx context.Context, client client.Client, graph *mcv1alpha3.GMConnector) error {
```

Comment:
Hi @KfreeZ, I notice that there is `client.Client` in `r`, it's unnecessary to pass `client` into this function.